nixos/modules/services/matrix/synapse.nix

```
@@ -105,6 +105,13 @@ let
         SYSLOG_IDENTIFIER = logName;
       };
     });
+
+  toIntBase8 = str:
+    lib.pipe str [
```

```suggestion
  toIntBase8 = str:
    if str == null then null else
    lib.pipe str [
```

Without this, `apply` for the `mode` will fail with "cannot coerce null to a string".

---

Left the `toIntBase8` function as is, and wrapped `toDecimalFilePermission` around it.